SourceX/controls/keymapper.hpp

```diff
@@ -0,0 +1,67 @@
+#pragma once
+
+#include <string>
+#include <functional>
+#include <vector>
+#include <unordered_map>
+
+namespace devilution {
+
+// The Keymapper maps keys to actions.
+class Keymapper final
+{
+public:
+	using SetConfigKeyFunction = std::function<void(const std::string &key, const std::string &value)>;
+	using GetConfigKeyFunction = std::function<std::string(const std::string &key)>;
+	using ActionIndex = std::size_t;
+
+	// Action represents an action that can be triggered using a keyboard shortcut.
+	class Action final
+	{
+	public:
+		// Can this action be triggered while the player is dead?
+		enum class IfDead
+		{
+			Allow,
+			Ignore,
+		};
+
+		Action(const std::string &name, int defaultKey, std::function<void()> action):
```

Comment:
Use a default argument, i.e. `IfDead ifDead = IfDead::Ignore`?

Reply:
Came across this [article](https://quuxplusone.github.io/blog/2020/04/18/default-function-arguments-are-the-devil/) recently that has convinced me not to use that feature. :)

Reply:
Interesting read.